ibis/expr/datatypes/core.py

```diff
@@ -59,6 +59,8 @@ def dtype_from_object(value, **kwargs) -> DataType:
                 raise TypeError(
                     f"Cannot construct an ibis datatype from python type `{value!r}`"
                 )
+        elif value is Any:
+            return json
```

Comment:
Investigate if this is still necessary and document why

Reply:
This isn't necessary, snowflake users will have to annotation with `json` explicitly for now.